app/components/image_input/image_input.js

```diff
@@ -0,0 +1,78 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+  static targets = [
+    'input',
+    'filename',
+    'thumbnail',
+    'link',
+    'selectedImage',
+    'emptyState',
+  ];
+
+  connect() {
+    const files = this.element.querySelector('input').files;
+
+    if (files.length > 0) {
+      this.updateFilePreview(files[0]);
+    }
+  }
+
+  showFilePicker() {
+    this.inputTarget.click();
+  }
+
+  handleChange(event) {
+    this.updateFilePreview(event.target.files[0]);
+    this.element.focus();
+  }
+
+  dropFile(event) {
+    event.preventDefault();
+
+    if (event.dataTransfer.files.length <= 0) {
+      return;
+    }
+
+    const file = event.dataTransfer.files[0];
+
+    if (!file.type.startsWith('image/')) {
+      return;
+    }
+
+    this.inputTarget.files = event.dataTransfer.files;
+    this.updateFilePreview(event.dataTransfer.files[0]);
+    this.element.focus();
+  }
+
+  updateFilePreview(file) {
```

Comment:
Call `revokeObjectURL` for the currently selected URL before creating a new one!